lua/telescope/builtin.lua

```diff
@@ -950,4 +950,79 @@ builtin.keymaps = function(opts)
   }):find()
 end
 
+builtin.tags = function(opts)
+  opts = opts or {}
+
+  if not vim.loop.fs_open('tags', "r", 438) then
+    print('Tags file does not exists. Create one with ctags -R')
+    return
+  end
+
+  local fd = assert(vim.loop.fs_open('tags', "r", 438))
+  local stat = assert(vim.loop.fs_fstat(fd))
+  local data = assert(vim.loop.fs_read(fd, stat.size, 0))
+  assert(vim.loop.fs_close(fd))
+
+  local lines = vim.split(data, '\n')
+
+  pickers.new(opts,{
+    prompt = 'Tags',
+    finder = finders.new_table {
+      results = lines,
+      entry_maker = make_entry.gen_from_ctags(opts),
+    },
+    previewer = previewers.ctags.new(opts),
+    sorter = conf.generic_sorter(opts),
+    attach_mappings = function(prompt_bufnr, map)
+      local update_lnum = actions._transform_action(function(prompt_bufnr)
+        local selection = actions.get_selected_entry(prompt_bufnr)
+        local strip_endline = string.gsub(selection.scode, '[$]$', '')
+        local strip_doubleslash = string.gsub(strip_endline, [[\\]], [[\]])
+        local strip_c_comment = string.gsub(strip_doubleslash, [[\/]], [[/]])
+
+        local cmd = 'grep -Frn \'' .. strip_c_comment .. '\' ' .. selection.filename
+        local output = vim.split(utils.get_os_command_output(cmd), ':')
```

Comment:
what's an example `scode` that you have in your ctags?

I thought `scodes` could almost just be `vim.fn.execute(scode)` or similar. I don't remember thougyh.

Reply:
Works with `vim.fn.search` but not for all `scode` i have. E.g. an example where it does not work is when i have an c comment at the end of the line. Scode would be:
```
ClkClientWin, ClkRootWin, ClkLast }; \/* clicks *\/$
```

I thought i could just unescape the `\/` but that wasn't enough and escaping the * as well. I had to also remove the $ sign at the end. I think it is good now.

Thanks for the insight.